core/vibes/soul/lib/streamable.tsx

```diff
@@ -2,6 +2,15 @@ import { Suspense, use } from 'react';
 
 export type Streamable<T> = T | Promise<T>;
 
+export function all<T extends readonly unknown[] | []>(
+  streamables: T,
+): Streamable<{ -readonly [P in keyof T]: Awaited<T[P]> }> {
+  return streamables.some((streamable) => streamable instanceof Promise)
+    ? Promise.all(streamables)
```

Comment:
Ideally this would be `Streamable.all`, but we can't namespace it like that in TS. Returns `Promise.all(streamables)` if at least one of the `streamables` is a promise, otherwise simply returns the `streamables` array.

This avoids a flash of loading indicator(s) when a component with multiple resolved streamable props that are combined using `Promise.all` is re-rendered on the client.

Reply:
This is something I was waiting to introduce. The reason is that we can't use `Promise.all` in client components because we can't create new promises in client components and call `use` as React will blow up. I would like to implement a `Streamable.all` utility that works in the client and the server, but to do that we need to be smart about using `Promise.all` and `use` depending on the context. Also, perhaps the answer isn't `Streamable.all` but a `<Stream>` variation that can take in many streamables (or a overload on the existing `<Stream>`)

Reply:
> This avoids a flash of loading indicator(s) when a component with multiple resolved streamable props that are combined using Promise.all is re-rendered on the client.

This shouldn't be happening 🤔